pkg/gui/presentation/commits.go

```diff
@@ -312,9 +312,17 @@ func displayCommit(
 	bisectInfo *git_commands.BisectInfo,
 	isYouAreHereCommit bool,
 ) []string {
-	hashColor := getHashColor(commit, diffName, cherryPickedCommitHashSet, bisectStatus, bisectInfo)
 	bisectString := getBisectStatusText(bisectStatus, bisectInfo)
 
+	hashString := ""
+	hashColor := getHashColor(commit, diffName, cherryPickedCommitHashSet, bisectStatus, bisectInfo)
+	hashLength := common.UserConfig.Gui.CommitHashLength
+	if hashLength > len(commit.Hash) || hashLength < 1 {
+		hashString = hashColor.Sprint(commit.ShortHash())
+	} else {
+		hashString = hashColor.Sprint(commit.Hash[:hashLength])
+	}
+
```

Comment:
Nitpick: I'd set hashString to the uncolored string, and use `hashColor.Sprint` in the append call below as before. Seems more logical to me, somehow.

Reply:
I did hashColor.Sprint there to be consistent with most of the rest of that function where Sprint is used to build the strings.
https://github.com/jesseduffield/lazygit/blob/961c3bc9212bd6223b3280e44ddec625e636bb7d/pkg/gui/presentation/commits.go#L326-L399

I like this format and would actually go further and make the few strings that were built inside the `cols` construction match the rest. 389a8c965d97b24f62ae231cbff9569b546851df.

The `cols` constructions ends up looking like this
https://github.com/jesseduffield/lazygit/blob/389a8c965d97b24f62ae231cbff9569b546851df/pkg/gui/presentation/commits.go#L392-L405

This commit doesn't change the functionality at all but I think the consistency makes it more readable. I haven't written in go before so maybe there was a reason it was like that. No need to include this commit in the pull, just a suggestion.

Reply:
Honestly I don't care too much about which way the code is written here, so I'm ok with your refactoring, but I don't really think it's necessary. One thing to be aware of is that now you are unconditionally adding divergence and description, whereas previously they were only added when needed. So now you are relying on the remove-empty-columns mechanism to remove them again; this works, but is unnecessary work.

My point about the hashString was that I don't like how the logic of assembling the string is mixed with coloring it. Not a huge deal either, I think we're talking about all this for longer than justified.

Reply:
If I understand correctly you'd prefer something like
```go
	cols = append(
		cols,
		hashColor.Sprint(divergenceString),
		hashColor.Sprint(hashString),
		// etc.
	)
```

I think I agree I prefer that logic as well, but like I said I did it the way I did because all of the rest of the strings in that function are colored and formatted where they are assembled and it felt weird to have just the one be colored there.